components/arm/client.go

```diff
@@ -13,33 +13,42 @@ import (
 
 	rprotoutils "go.viam.com/rdk/protoutils"
 	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/resource"
 	"go.viam.com/rdk/spatialmath"
 )
 
 var errArmClientModelNotValid = errors.New("unable to retrieve a valid arm model from arm client")
 
 // client implements ArmServiceClient.
 type client struct {
+	resource.Named
+	resource.TriviallyReconfigurable
 	name   string
-	conn   rpc.ClientConn
 	client pb.ArmServiceClient
 	model  referenceframe.Model
 	logger golog.Logger
 }
 
 // NewClientFromConn constructs a new Client from connection passed in.
-func NewClientFromConn(ctx context.Context, conn rpc.ClientConn, name string, logger golog.Logger) Arm {
-	c := pb.NewArmServiceClient(conn)
-	// TODO: DATA-853 requires that this support models being changed on the fly, not just at creation
-	// TODO: RSDK-882 will update this so that this is not necessary
+func NewClientFromConn(ctx context.Context, conn rpc.ClientConn, name resource.Name, logger golog.Logger) (Arm, error) {
+	pbClient := pb.NewArmServiceClient(conn)
+	// TODO(DATA-853): requires that this support models being changed on the fly, not just at creation
+	// TODO(RSDK-882): will update this so that this is not necessary
 	r := robotpb.NewRobotServiceClient(conn)
-	return &client{
-		name:   name,
-		conn:   conn,
-		client: c,
-		model:  getModel(ctx, r, name),
+	model, hasModel, err := getModel(ctx, r, name.ShortNameForClient())
+	if err != nil {
+		return nil, err
+	}
+	c := &client{
+		Named:  name.AsNamed(),
+		name:   name.ShortNameForClient(),
```

Comment:
[q] This seems to be a pattern of having both `Named` and `name` in struct.  Wondering what the thought is for this

Reply:
responded elsewhere. ideally these would be generic anyway. see the interface for Named.